connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -545,6 +545,67 @@ else if (!configState.connectors().contains(connName))
     }
 
     @Override
+    public synchronized void restartConnector(final String connName, final Callback<Void> callback) {
+        addRequest(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                if (!configState.connectors().contains(connName)) {
+                    callback.onCompletion(new NotFoundException("Unknown connector: " + connName), null);
```

Comment:
In some config related methods we use `checkConfigSynced` to validate that we at least think we have a valid set of configs and assignment that are likely to be in sync with other workers. I don't think that method will work quite right here, but do we want something similar? I feel that using `configState` before checking that may lead to confusing results if any nodes are out of sync, though I haven't thought through what would be useful feedback in that case...